tests/platform_tests/verify_dut_health.py

```diff
@@ -0,0 +1,150 @@
+    if failed:
+        raise RebootHealthError("Interface check failed, not all interfaces are up. Failed: {}".format(failed))
+
+    # Skip this step for virtual testbed - KVM testbed has transeivers marked as "Not present"
+    # and the DB returns an "empty array" for "keys TRANSCEIVER_INFO*"
+    if duthost.facts['platform'] == 'x86_64-kvm_x86_64-r0':
+        return
+
+    logging.info("Check whether transceiver information of all ports are in redis")
+    xcvr_info = duthost.command("redis-cli -n 6 keys TRANSCEIVER_INFO*")
+    parsed_xcvr_info = parse_transceiver_info(xcvr_info["stdout_lines"])
+    interfaces = conn_graph_facts["device_conn"][duthost.hostname]
+    for intf in interfaces:
+        if intf not in parsed_xcvr_info:
+            raise RebootHealthError("TRANSCEIVER INFO of {} is not found in DB".format(intf))
+
+
+@handle_test_error
+def check_neighbors(duthost, tbinfo):
+    """
+    Perform a BGP neighborship check.
+    """
+    logging.info("Check BGP neighbors status. Expected state - established")
+    bgp_facts = duthost.bgp_facts()['ansible_facts']
+    mg_facts  = duthost.get_extended_minigraph_facts(tbinfo)
+
+    for value in bgp_facts['bgp_neighbors'].values():
+        # Verify bgp sessions are established
+        if value['state'] != 'established':
+            raise RebootHealthError("BGP session not established")
+        # Verify locat ASNs in bgp sessions
+        if(value['local AS'] != mg_facts['minigraph_bgp_asn']):
+            raise RebootHealthError("Local ASNs not found in BGP session.\
+                Minigraph: {}. Found {}".format(value['local AS'], mg_facts['minigraph_bgp_asn']))
+    for v in mg_facts['minigraph_bgp']:
+        # Compare the bgp neighbors name with minigraph bgp neigbhors name
+        if(v['name'] != bgp_facts['bgp_neighbors'][v['addr'].lower()]['description']):
+            raise RebootHealthError("BGP neighbor's name does not match minigraph.\
+                Minigraph: {}. Found {}".format(v['name'], bgp_facts['bgp_neighbors'][v['addr'].lower()]['description']))
+        # Compare the bgp neighbors ASN with minigraph
+        if(v['asn'] != bgp_facts['bgp_neighbors'][v['addr'].lower()]['remote AS']):
+            raise RebootHealthError("BGP neighbor's ASN does not match minigraph.\
+                Minigraph: {}. Found {}".format(v['asn'], bgp_facts['bgp_neighbors'][v['addr'].lower()]['remote AS']))
+
+
+@handle_test_error
+def verify_no_coredumps(duthost, pre_existing_cores):
+    coredumps_count = duthost.shell('ls /var/core/ | wc -l')['stdout']
+    if int(coredumps_count) > int(pre_existing_cores):
+        raise RebootHealthError("Core dumps found. Expected: {} Found: {}".format(pre_existing_cores,\
+            coredumps_count))
+
+
+def wait_until_uptime(duthost, post_reboot_delay):
+    logging.info("Wait until DUT uptime reaches {}s".format(post_reboot_delay))
+    while duthost.get_uptime().total_seconds() < post_reboot_delay:
+        time.sleep(1)
```

Comment:
We can optimize the sleep according to the gap between the current uptime and the expected uptime. With another PR.